src/Corax/Queries/Suggestions.DistanceProviders.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Voron;
+
+namespace Corax.Queries;
+
+public enum StringDistanceAlgorithm
+{
+    None,
+    NGram,
+    JaroWinkler,
+    Levenshtein
+}
+
+public struct LevenshteinDistance : IStringDistance
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public float GetDistance(Slice target, Slice other)
+    {
+        //Taken from RavenDB
```

Comment:
?? Where

Reply:
It's here:

https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Server/Documents/Indexes/Persistence/Lucene/Suggestions/LevenshteinDistance.cs

same like other distance algorithms:
https://github.com/ravendb/ravendb/tree/v5.3/src/Raven.Server/Documents/Indexes/Persistence/Lucene/Suggestions

It would be good to have it implemented as generic class - `LevenshteinDistance<T>`. So for Corax we'd have `LevenshteinDistance<Slice>` while for Lucene `LevenshteinDistance<string>`.

CC @redknightlois

Reply:
Note that this isn't from RavenDB, but from Lucene originally, and should be noted accordingly. Preferrably with a link to the source